### DISCUSSION TITLE
front socket object connected  always is false

### DISCUSSION THREAD

**Discussion by LingerW** [socketio/socket.io#4746] [Q&A]
front socket object connected print  always is false 
socket.io-client version is ^4.7.0

server is netty-socketio 2.0.2

<img width="217" alt="image" src="https://github.com/socketio/socket.io/assets/45159104/d814cc09-32e1-4a4c-86a1-b6230b48a005">

<img width="991" alt="image" src="https://github.com/socketio/socket.io/assets/45159104/83f7f676-6a08-4ef5-ae7a-4663e980fb5c">
<img width="681" alt="image" src="https://github.com/socketio/socket.io/assets/45159104/60ae6e3b-217c-4648-a2c2-fbc61b2c0333">

websocket is ready but front socket object connected is always false and i can not use it to emit any event 

what can i do to solve it

**Comment by darrachequesne** [comment]
Hi! Did you check our guide with React: https://socket.io/how-to/use-with-react?

**Reply by LingerW** [reply]
of course! but i use Manager mode to create websocket connect
when i use io function to create socket object, the socket object connected is true but it can not work in useEffect hook;

when i use socket.io-client version 2.4.0, it can work in hooks;
